src/shogun2-web/src/main/java/de/terrestris/shogun2/rest/AbstractRestController.java

```diff
@@ -49,25 +58,48 @@
 	 * @return
 	 */
 	@RequestMapping(value = "/{id}", method = RequestMethod.GET)
-	public ResponseEntity<E> get(@PathVariable int id) {
-		E entity = this.service.findById(id);
-		return new ResponseEntity<E>(entity, HttpStatus.OK);
+	public ResponseEntity<E> findById(@PathVariable Integer id) {
+
+		try {
+			E entity = this.service.findById(id);
+			LOG.trace("Found " + entity.getClass().getSimpleName()
+					+ " with ID " + entity.getId());
+			return new ResponseEntity<E>(entity, HttpStatus.OK);
+		} catch (Exception e) {
+			LOG.error("Error finding entity with id " + id + ": "
+					+ e.getMessage());
+			return new ResponseEntity<E>(HttpStatus.NOT_FOUND);
+		}
 	}
 
 	/**
-	 * Create an entity.
+	 * Create/save an entity.
 	 *
 	 * @param entity
 	 * @return
 	 */
 	@RequestMapping(method = RequestMethod.POST)
-	public ResponseEntity<E> create(@RequestBody E entity) {
+	public ResponseEntity<E> save(@RequestBody E entity) {
+
+		final String simpleClassName = entity.getClass().getSimpleName();
+		final String errorMessagePrefix = "Error when saving entity of type "
+				+ simpleClassName + ": ";
+
+		// ID value MUST be null to assure that
+		// saveOrUpdate will save and not update
+		final Integer id = entity.getId();
+		if (id != null) {
+			LOG.error(errorMessagePrefix + "ID value is set to " + id
+					+ ", but MUST be null");
+			return new ResponseEntity<E>(HttpStatus.BAD_REQUEST);
+		}
+
 		try {
-			E created = this.service.saveOrUpdate(entity);
-			LOG.debug("Create " + entity.getClass() + " with body " + entity);
-			return new ResponseEntity<E>(created, HttpStatus.CREATED);
+			entity = this.service.saveOrUpdate(entity);
```

Comment:
Don't we now know that this will not be a CREATE? Is there a dedicazed `save` in the service?